File: 3094-minimum-number-of-operations-to-make-array-empty/minimum-number-of-operations-to-make-array-empty.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        cnt = 0 

      
        counter = Counter(nums)
        for v, k in counter.items():
            if k==1:
                return -1
            if k%2==0 and k%3==0:
                cnt += k//3
            elif k%3==2 and k%2==0:
                cnt += k//3 + 1
            elif  k%3==1 and k%2==0:
                cnt += k//3 + 1 
            elif  k%3==0:
                cnt += k//3
            elif  k%3==1 and k%2!=0:
                cnt += k//3 + 1 
            elif  k%3==2 and k%2!=0:
                cnt += k//3 + 1 
            
        return cnt 
                
                
                

File: 3094-minimum-number-of-operations-to-make-array-empty/minimum-number-of-operations-to-make-array-empty.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int cnt = 0 ;
        unordered_map<int,int>mp ; 
        for(int i : nums){
            mp[i]++;
        }
        for(auto it : mp){
            if(it.second==1) return -1 ;
            else{
                int x = it.second ;
                if(x%3==0) 
                cnt += x/3;
                else
                cnt += x/3 + 1 ;
                
            }
        }
        return cnt ;
    }
};